.github/workflows/integrate.yml

```diff
@@ -156,6 +156,7 @@ jobs:
         run: |
           # Compare the integration branch with the target branch
           CHANGED_FILES=$(git diff integration..main --name-only | wc -l)
+          HAS_NON_LOCK_CHANGES=[ "$CHANGED_FILES" -gt 0 ] && ([ "$CHANGED_FILES" -gt 1 ] || [ $(git diff integration..main --name-only) != "Cargo.lock" ])
```

Comment:
As discussed, I think that checking `$CHANGED_FILES -gt 1` is sufficient, but the extra checking certainly doesn't hurt.